packages/devtools_app/lib/src/screens/vm_developer/object_viewport.dart

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:vm_service/vm_service.dart' hide Stack;
+
+import '../../primitives/history_manager.dart';
+import '../../shared/common_widgets.dart';
+import '../../shared/history_viewport.dart';
+import 'object_inspector_view_controller.dart';
+import 'vm_class_display.dart';
+import 'vm_developer_common_widgets.dart';
+import 'vm_object_model.dart';
+
+/// Displays the VM information for the currently selected object in the
+/// program explorer.
+class ObjectViewport extends StatelessWidget {
+  const ObjectViewport({
+    Key? key,
+    required this.controller,
+    this.initialObject,
+  }) : super(key: key);
+
+  final ObjectInspectorViewController controller;
+
+  final VmObject? initialObject;
+
+  @override
+  Widget build(BuildContext context) {
+    return HistoryViewport<VmObject>(
+      history: controller.objectHistory,
+      controls: [
+        ToolbarAction(
+          icon: Icons.refresh,
+          onPressed: () {
+            controller.refreshObject();
+          },
+          tooltip: 'Refresh',
+        )
+      ],
+      generateTitle: getViewportTitle,
+      contentBuilder: (context, _) {
+        return ValueListenableBuilder<bool>(
+          valueListenable: controller.refreshing,
+          builder: (context, refreshing, _) {
+            Widget child;
+
+            if (refreshing) {
+              child = const CenteredCircularProgressIndicator();
+            } else {
+              final currentObject = controller.objectHistory.current.value;
+              child = currentObject == null
+                  ? const SizedBox.shrink()
+                  : buildObjectScreen(currentObject);
+            }
+
+            return Expanded(child: child);
+          },
+        );
+      },
+    );
+  }
+}
+
+@visibleForTesting
+String getViewportTitle(VmObject? object) {
```

Comment:
nit: avoid starting method names with get: https://dart.dev/guides/language/effective-dart/design#avoid-starting-a-method-name-with-get

Replace with _viewportTitle or _generateViewportTitle, or similar

Reply:
Thanks for sharing!